packages/react-router/src/router.ts

```diff
@@ -1903,10 +1907,11 @@ export class Router<
                       // we can use the options
                       await lazyPromise
 
-                      checkLatest()
+                      // checkLatest()
```

Comment:
Not sure why I had to comment this out here. TBH, I don't fully understand what the purpose of the controlled promises are and their function.

Reply:
what happens if you do not comment this?

Reply:
If I restore it to what it originally was:
```
...
                      // we can use the options
                      await lazyPromise

                      checkLatest()

                      // Kick off the loader!
                      loaderPromise = route.options.loader?.(loaderContext)
```

This is the output:
[output.md](https://github.com/user-attachments/files/15754137/output.md)